crates/typst-syntax/src/lexer.rs

```diff
@@ -27,9 +27,9 @@ pub(super) struct Lexer<'s> {
 pub(super) enum LexMode {
     /// Text and markup.
     Markup,
-    /// Math atoms, operators, etc.
+    /// Math mode, i.e. those in `$ .. $`.
     Math,
-    /// Keywords, literals and operators.
+    /// Code mode, e.g. `#let a = 1`, `#{ ... }`.
```

Comment:
This is a bit misleading, since `#let ..` is an embedded markup expression. I'd prefer if the doc comments were reverted.